libraries/chain/webassembly/crypto.cpp

```diff
@@ -108,94 +107,83 @@ namespace eosio { namespace chain { namespace webassembly {
-      using error_code = eosio::chain::webassembly::error_codes::crypto;
-
       bytes bg1_point(g1_point.data(), g1_point.data() + g1_point.size());
       bytes bscalar(scalar.data(), scalar.data() + scalar.size());
 
       auto maybe_err = fc::alt_bn128_mul(bg1_point, bscalar);
       if(std::holds_alternative<fc::alt_bn128_error>(maybe_err)) {
-         return error_code::fail;
+         return return_code::failure;
       }
 
       const auto& res = std::get<bytes>(maybe_err);
 
       if( result.size() < res.size() )
-         return error_code::fail;
+         return return_code::failure;
 
       std::memcpy( result.data(), res.data(), res.size() );
-      return error_code::none;
+      return return_code::success;
    }
 
-   int32_t interface::alt_bn128_pair(span<const char> g1_g2_pairs, bool* result) const {
-      using error_code = eosio::chain::webassembly::error_codes::crypto;
-
+   int32_t interface::alt_bn128_pair(span<const char> g1_g2_pairs) const {
       bytes bg1_g2_pairs(g1_g2_pairs.data(), g1_g2_pairs.data() + g1_g2_pairs.size());
 
       auto checktime = [this]() { context.trx_context.checktime(); };
-      auto maybe_err = fc::alt_bn128_pair(bg1_g2_pairs, checktime);
-      if(std::holds_alternative<fc::alt_bn128_error>(maybe_err)) {
-         return error_code::fail;
+      auto res = fc::alt_bn128_pair(bg1_g2_pairs, checktime);
+      if(std::holds_alternative<fc::alt_bn128_error>(res)) {
+         return return_code::failure;
       }
 
-      *result = std::get<bool>(maybe_err);
-      return error_code::none;
+      return !std::get<bool>(res); 
    }
 
    int32_t interface::mod_exp(span<const char> base,
                               span<const char> exp,
                               span<const char> modulus,
                               span<char> out) const {
-      using error_code = eosio::chain::webassembly::error_codes::crypto;
-
       bytes bbase(base.data(), base.data() + base.size());
       bytes bexp(exp.data(), exp.data() + exp.size());
       bytes bmod(modulus.data(), modulus.data() + modulus.size());
 
       auto maybe_err = fc::modexp(bbase, bexp, bmod);
       if(std::holds_alternative<fc::modular_arithmetic_error>(maybe_err)) {
-         return error_code::fail;
+         return return_code::failure;
       }
 
       const auto& res = std::get<bytes>(maybe_err);
 
       if( out.size() < res.size() )
-         return error_code::fail;
+         return return_code::failure;
 
       std::memcpy( out.data(), res.data(), res.size() );
-      return error_code::none;
+      return return_code::success;
    }
 
    int32_t interface::blake2_f( uint32_t rounds,
                                 span<const char> state,
                                 span<const char> message,
                                 span<const char> t0_offset,
                                 span<const char> t1_offset,
-                                bool final,
+                                int32_t final,
```

Comment:
Should there be special logic to handle values other than 0 or 1 provided by the caller? What is the protocol claiming to do in that case?